### PR TITLE
tablex: Add spairs (alternative of pairs that returns sorted table)

### PR DESCRIPTION
**Pull request by jalhund**

### CODE REVIEW

File: tablex.lua [hunk 1]
```diff
@@ -537,4 +537,28 @@ function tablex.ripairs(t)
 	return _ripairs_iter, t, #t + 1
 end
 
+-- works like pairs, but returns sorted table
+function tablex.spairs(t, fn)
+	local keys = {}
+	for k in pairs(t) do
```
Comment (1bardesign) [inline tablex.lua:543]: this should probably just be `tablex.keys`

File: tablex.lua [hunk 2]
```diff
@@ -537,4 +537,28 @@ function tablex.ripairs(t)
...
+		tablex.push(keys, k)
+	end
+
+	if fn then
+		table.sort(keys, function(a,b) return fn(t, a, b) end)
```
Comment (1bardesign) [inline tablex.lua:548]: is there a reason the signature for the function provided isn't just `f(a, b)` - seems like it'd be expected?

**Review by 1bardesign — CHANGES_REQUESTED** [review]
a few things to touch up :pray: i like the idea though, we may use it inside `pretty` to make the ordering consistent!

File: tablex.lua (hunk 2 above)
Comment (jalhund) [reply tablex.lua:548]: Sorry, I don't understand what do I need to change. Could you please write in more detail?

File: tablex.lua [hunk 3]
```diff
@@ -537,4 +537,28 @@ function tablex.ripairs(t)
...
+	else
+		-- sort by keys if no function passed
+		table.sort(keys)
```
[the remark this answers is not in the document]
Comment (jalhund) [reply tablex.lua:551]: I just want say that this will add dependency from sort module to tablex module. Implement?

**Comment by 1bardesign** [comment]
Don't worry, I sorted it out to save the back and forward. One change was made; the `less` function passed doesn't get a reference to the table, as it can be accessed through a closure if needed (and this function is slow enough that I think it's not worth worrying about the overhead of a closure), and it means that the default `less` and any other existing comparison functions dont need wrapping to accept a table.

Sorry if that's a pain for you and thanks for the contribution!

**Comment by jalhund** [comment]
Sorry you had to fix my code and not me. I didn't do it on purpose, I just didn't fully understand what was meant. Now I see your new comments and code and understand. You are great lua programmer, thank you

**Comment by 1bardesign** [comment]
:+1: :pray: no worries mate

**Pull request merged**